tools/ci_build/compile_triton.py

```diff
@@ -0,0 +1,179 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+# --------------------------------------------------------------------------
+
+import argparse
+import importlib.util
+import os
+import shutil
+
+import triton
+
+"""
+These files are based on onnxruntime/onnxruntime root dir.
+and should implement get_function_table() with function table format:
+
+function_table = [
+        {'name': xx,
+         'group': yy,
+         'func': func,
+         'sig': sig,
+         'kwargs': kwargs
+        }
+]
+
+"""
+_TRITON_SCRIPT_FILES = [
+    "core/providers/rocm/math/softmax.py",
+]
+
+
+def compile(function_table, out_dir):
+    def compile_one(func, sig, **kwargs):
+        ret = triton.compile(func, signature=sig, **kwargs)
+        return ret
+
+    metadata = []
+    for func_desc in function_table:
+        name = func_desc["name"]
+        group = func_desc["group"]
+        sig = func_desc["sig"]
+        func = func_desc["func"]
+        kwargs = func_desc["kwargs"]
+
+        print("compile func: ", func_desc)
+
+        ret = compile_one(func, sig, **kwargs)
+
+        compile_res = {}
+        compile_res["name"] = name
+        compile_res["group"] = group
+        compile_res["func_name"] = ret.metadata["name"]
+        compile_res["num_warps"] = ret.metadata["num_warps"]
+        compile_res["shared"] = ret.metadata["shared"]
+        if "constants" in kwargs:
+            compile_res["constants"] = kwargs["constants"]
+
+        # move tmp hsaco file into current dir
+        lib_name = f"{name}.hsaco"
+        shutil.copyfile(ret.asm["hsaco_path"], f"{out_dir}/{lib_name}")
+        compile_res["lib_file"] = lib_name
+        metadata.append(compile_res)
+
+    return metadata
+
+
+def convert_lib_to_obj(lib_file, out_dir):
+    obj_file = lib_file.split(".")[0] + ".o"
+    command = f"cd {out_dir}; objcopy -I binary -O elf64-x86-64 -B i386:x86-64 {lib_file} {obj_file}; cd -"
+    ret = os.system(command)
+
+    if ret != 0:
+        raise Exception(f"exec convert command: {command} failed.")
+    # check file exist
+    if not os.path.exists(f"{out_dir}/{obj_file}"):
+        raise Exception(f"the output file not exist, after exec comamnd: {command}")
+
+    return obj_file
+
+
+def archive_obj_files(obj_files, out_dir, out_obj_file):
+    obj_files = " ".join(obj_files)
+    command = f"cd {out_dir}; ar rcs {out_obj_file} {obj_files}; cd -"
+    ret = os.system(command)
+
+    if ret != 0:
+        raise Exception(f"exec convert command: {command} failed.")
+    # check file exist
+    if not os.path.exists(f"{out_dir}/{out_obj_file}"):
+        raise Exception(f"the output file not exist, after exec comamnd: {command}")
```

Comment:
Use `OBJECT` library, you don't need this and whole-archive linking.

Reply:
it needs --whole-archive, because the obj file is treated as binary data, don't have symbols that  librocm.so is depend on, so if not using --whole-archive, the obj will not be combined into librocm.so.

Reply:
> Use OBJECT library,